src/text-field.jsx

```diff
@@ -16,7 +16,7 @@ const ContextPure = require('./mixins/context-pure');
  * @returns True if the string provided is valid, false otherwise.
  */
 function isValid(value) {
-  return value || value === 0;
+  return Boolean(value || value === 0);
```

Comment:
Indeed (http://adripofjavascript.com/blog/drips/the-difference-between-boolean-objects-and-boolean-primitives-in-javascript.html)
I would use @luisrudge :+1: